### PR TITLE
fix: [PIE-4746]: input set values retaining on yaml visual switch

### PR DESCRIPTION
**Pull request by vikrant-harness**
### Summary

fix: [PIE-4746]: input set values retaining on yaml visual switch

#### Screenshots

https://user-images.githubusercontent.com/106532291/183256024-046647b0-c3a9-43df-8456-25d99107b97d.mov

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress: `trigger cypress`
- Fix Prettier: `fix prettier`
</details>

[PIE-4746]: https://harness.atlassian.net/browse/PIE-4746?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by diptiman-adak** [comment]
Quality gate Quality gate: **_Passed_** 
 


            
 No issues Everything looks good
 [Details](https://app.embold.io/organization/gh/harness/repositories/e864badd0b95ab81c060b15b85f15701/pull-request/10439)

**Comment by vikrant-harness** [comment]
retrigger titecheck

**Comment by vikrant-harness** [comment]
retrigger titlecheck

**Review by sonaligoyal-harness — APPROVED** [review]

**Comment by vikrant-harness** [comment]
trigger cypress

**Review by PratyushGar — DISMISSED** [review]

**Review by bhavya-sinha — DISMISSED** [review]

**Review by bhavya-sinha — APPROVED** [review]

**Review by YounixIO — APPROVED** [review]

File: src/modules/70-pipeline/components/InputSetForm/FormikInputSetForm.tsx [hunk 1]
```diff
@@ -306,6 +310,11 @@ export default function FormikInputSetForm(props: FormikInputSetFormProps): Reac
     filePath: defaultTo(inputSet.gitDetails?.filePath, filePath)
   }
 
+  React.useEffect(() => {
+    const initialValues = getPipelineData()
```
Comment (chetanvekariya-harness) [inline src/modules/70-pipeline/components/InputSetForm/FormikInputSetForm.tsx:314]: Do not need to make the call again. Use **init.pipeline**. Use **init** in dependency.

File: src/modules/70-pipeline/components/InputSetForm/FormikInputSetForm.tsx [hunk 2]
```diff
@@ -306,6 +310,11 @@ export default function FormikInputSetForm(props: FormikInputSetFormProps): Reac
...
+    formikRef.current?.setFieldValue('pipeline', initialValues.pipeline)
+  }, [inputSet, isEdit, resolvedPipeline])
```
Comment (chetanvekariya-harness) [inline src/modules/70-pipeline/components/InputSetForm/FormikInputSetForm.tsx:316]: inputSet and isEdit are not used anywhere in the function. Do we need those in dependency list ?

File: src/modules/70-pipeline/components/InputSetForm/FormikInputSetForm.tsx (hunk 1 above)
Comment (vikrant-harness) [reply src/modules/70-pipeline/components/InputSetForm/FormikInputSetForm.tsx:314]: Tried this out but the object is not getting updated the issue still persists after that

File: src/modules/70-pipeline/components/InputSetForm/FormikInputSetForm.tsx (hunk 2 above)
Comment (vikrant-harness) [reply src/modules/70-pipeline/components/InputSetForm/FormikInputSetForm.tsx:316]: the getPipeline function is dependent on these fields.

**Review by bhavya-sinha — APPROVED** [review]

**Pull request merged**